### PR TITLE
Update k8s ingress image

### PR DESCRIPTION
**Pull request by alexvy86**
## Description

- Update the Nginx ingress controller deployment to use a Helm chart with the most recent image (the previous one was pretty old and completely broken after the AKS cluster upgrade from k8s 1.19 to 1.23) instead of manual steps to deploy it. This new approach is what is currently deployed to the AKS cluster and confirmed to be allowing traffic into the other services in there.
  - I put the ingress controller for each environment in corresponding `prod` and `ppe` namespaces in the k8s cluster, as a starting point to eventually move all the components to the specific namespaces as a best practice. The ingress controllers still handle traffic for `Ingress` resources in any namespace, including the `default` one we currently use for most components, as long as the `Ingress` resources are correctly configured to use the `IngressClass` resources that the ingress controllers are tied to.
- Update several manifests (particularly historian and routerlicious) to work with the updated ingress controller (replace an annotation with a new property, update some annotation namespaces).
- Update/cleanup some of the documentation.

## Reviewer Guidance

The value of `dhParam` in the two new manifests is public information shared between client and server when establishing a TLS connection, not a secret that should be out of source control.

### CODE REVIEW

**Comment by hedasilv** [comment]
+ @sharmakhushboo and @zhajie who have recently worked on areas related to nginx

File: server/charts/historian/templates/historian-ingress.yaml [hunk 1]
```diff
@@ -9,23 +9,23 @@ metadata:
     release: "{{ .Release.Name }}"
     heritage: "{{ .Release.Service }}"
   annotations:
-    kubernetes.io/ingress.class: {{ .Values.historian.ingressClass }}
-    ingress.kubernetes.io/ssl-redirect: "true"
+    nginx.ingress.kubernetes.io/ssl-redirect: "true"
     nginx.org/client-max-body-size: "100m"
-    ingress.kubernetes.io/proxy-body-size: "100m"
+    nginx.ingress.kubernetes.io/proxy-body-size: "100m"
```
Comment (sharmakhushboo) [inline server/charts/historian/templates/historian-ingress.yaml:14]: Curious. Is this change really required?
In FRS, we are using nginx version v1.1.2 and we didn't have to make any changes to the service ingress files.

Comment (sharmakhushboo) [reply server/charts/historian/templates/historian-ingress.yaml:14]: Yes, we are using the prefix like this:
image

Comment (sharmakhushboo) [reply server/charts/historian/templates/historian-ingress.yaml:14]: Without the prefix, would this break existing nginx deployments?

Comment (alexvy86) [reply server/charts/historian/templates/historian-ingress.yaml:14]: > Yes, we are using the prefix like this:

That's in the Ingress resource, but what about the ingress controller? With v1.1.2 of that one, I'd expect that you're passing the `--annotations-prefix=ingress.kubernetes.io` flag to it, so that it looks for those annotations on the Ingress (and not the ones it would look for by default, with `nginx.ingress.kubernetes.io/`). Not sure if you're using a "manual" deployment (with manifests) like this repo seemed to have been doing before my changes, or also moved to the Helm chart, but in either case you're probably passing that flag to the controller, correct? If you are, then my understanding is that changing the annotation in the Ingresses without removing that flag from the controller will make it so the controller ignores the annotations. The upgrade path I would follow would be to first deploy the Ingresses with both versions of the annotations, then deploy the controller without the flag, and finally remove the annotations with the current prefix.

> I guess my question is, were you able to test your changes on an existing setup to make sure nothing breaks?

I did deploy these changes to our internal cluster and as far as I can tell, the Ingress Controller is working as expected. But to reiterate, I changed both the Ingresses and the ingress controller at the same time, so both would "agree" on which version of the annotation prefix was the one to use.

Comment (sharmakhushboo) [reply server/charts/historian/templates/historian-ingress.yaml:14]: We don't actually. The only args that we pass to the controller are below. And we don't pass any explicit annotations either.
image

Comment (alexvy86) [reply server/charts/historian/templates/historian-ingress.yaml:14]: Huh... then do we know if the annotations are actually being respected? I just noticed that [the `ssl-redirect` one defaults to `true` anyway as long as TLS is configured for the Ingress](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/#server-side-https-enforcement-through-redirect), which is our case, so we would see that behavior whether the annotation was there or not. The `proxy-body-size` one for rejecting requests over 100M seems a bit harder to test, but maybe it's worth doing it? Because if I had to guess, I'd say it's not being applied, because the controller must be looking for the newer prefix (`nginx.ingress.kubernetes.io`); and I'd expect the [default value of 1M](https://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size) to be the effective one right now.

So, given the annotations we use, the mismatch between the prefix expected by the controller and the one specified in the annotations would not result in any immediate breaking that I can think of; the only difference would be that requests between 1M and 100M in size would be rejected when in theory we don't want them to be.

Comment (sharmakhushboo) [reply server/charts/historian/templates/historian-ingress.yaml:14]: Hey Alex. As per my understanding, the annotations are being respected, but we are not using helm for nginx yet. So, if your changes are working fine in your AKS cluster, I think you can go ahead and merge these changes. :) 

also, about body-size, we use the below configuration to limit the size of client requests. It's been working for us so far.

`client_max_body_size                    100M; `

**Review by sharmakhushboo — APPROVED** [review]
